scripts/check-test-determinism.py

```
@@ -0,0 +1,915 @@
+    if suffix not in SCANNED_SUFFIXES:
+        return False
+    # Under Packages/, only scan files inside a Tests path segment.
+    if root == "Packages" and "/Tests/" not in ("/" + rel_posix):
+        return False
+    return True
+
+
+def scan_text(rel_posix: str, text: str) -> list[Finding]:
+    suffix = pathlib.PurePosixPath(rel_posix).suffix
+    raw_lines = text.splitlines()
+    code_lines = [_strip_comment(l, suffix) for l in raw_lines]
```

Ruff (0.15.17)

[error] Ambiguous variable name: `l`

(E741)